Assignment2/Logger/LogFactory.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace Logger
+{
+    
+  
+    public class LogFactory
+    {
+        private string LogFilePath { get; set; }
+        public BaseLogger CreateLogger(string className)
```

Comment:
Part of the assignment requirement was to enable `Nullable` and `TreatWarningsAsErrors` in `Logger.csproj`. If that would have been done, you would have seen that the code would have build errors forcing you to make the private string and the return type on the method nullable. Also remember that it is best practice to preface a private class field with an underscore (`_LogFilePath`)
```suggestion
        private string? _LogFilePath { get; set; }
        public BaseLogger? CreateLogger(string className)
```